src/Raven.Server/Documents/Indexes/Static/DynamicDictionary.cs

```diff
@@ -0,0 +1,593 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Raven.Server.Documents.Indexes.Static
+{
+    public class DynamicDictionary : IDictionary<object, object>, IOrderedEnumerable<object>
+    {
+        private readonly Dictionary<object, object> _dictionary;
+
+        public DynamicDictionary(Dictionary<object, object> dictionary)
+        {
+            _dictionary = dictionary;
+        }
+
+        public DynamicDictionary(IEnumerable<KeyValuePair<object, object>> dictionary)
+        {
+            _dictionary = (Dictionary<object, object>)dictionary;
```

Comment:
You need to construct a new dictionary here, you can't assume that this is a Dictionary

Reply:
done